src/ripple/app/ledger/Ledger.cpp

```diff
@@ -225,14 +225,14 @@ Ledger::Ledger (uint256 const& parentHash,
     loaded = true;
 
     if (info_.txHash.isNonZero () &&
-        !txMap_->fetchRoot (info_.txHash, nullptr))
+        !txMap_->fetchRoot (SHAMapHash{info_.txHash}, nullptr))
```

Comment:
I think those would be a step _forward_! :-)  However this is my second attempt at this re-typing.  My first attempt tried to do it rippled-wide and I found it was overly ambitious (error prone).  So this is phase 1 of a multi-step process to do it rippled-wide.  This phase is intentionally drawing the line at `SHAMap`'s borders.  Get something that can be compiled, tested and inspected, and then move on from there.